pgcontents/tests/test_pgcontents_api.py

```diff
@@ -163,6 +163,17 @@ def test_list_checkpoints_sorting(self):
             )
         )
 
+    # skip test as it is not satisfyable across supported notebook versions
+    def test_delete_non_empty_dir(self):
+        # ContentsManager has different behaviour in notebook 5.5+
```

Comment:
So, from reading the discussion on https://github.com/jupyter/notebook/pull/3108, it seems like the reason notebook now allows deleting non-empty directories is that they switched to using `send2trash` for deletions by default, which means it's possible for a user to recover a lost directory that they accidentally deleted.  Since deletions in pgcontents are still irreversible, my inclination would be to continue to test that we don't support deletion of nonempty directories when using  `PostgresContentsManager`. 

I think the easiest way to test that ensure that behavior is to make this method conditional on the value of `self.config.NotebookApp.contents_manager_class`. If it's a subclass of `PostgresContentsManager`, we should keep the test for the old behavior. For FileContentsManager we can just call `super()` to test that we haven't somehow broken the default behavior from upstream. For HybridContentsManager, the expected behavior would depend on how the sub-managers are configured, so the right test would probably just be to assert that we prevent deletions in the subdirectory owned by the postgres manager. 

Apologies for the somewhat complex feedback here: this suite is a bit convoluted, but since this is testing the code that controls whether/how we delete user data on Quantopian, I want to err on the side of caution. If you prefer, I can also open a PR against this branch with the changes I have in mind for the suite.